pkg/fqdn/cache.go

```diff
@@ -170,6 +168,10 @@ func NewDNSCacheWithLimit(limit int) *DNSCache {
 // ips may be an IPv4 or IPv6 IP. Duplicates will be removed.
 // ttl is the DNS TTL for ips and is a seconds value.
 func (c *DNSCache) Update(lookupTime time.Time, name string, ips []net.IP, ttl int) bool {
+	if c.minTTL > 0 && ttl < c.minTTL {
+		ttl = c.minTTL
```

Comment:
I just realised that it would be better to also keep a field with the original request TTL. This way, when we restore with a different minTTL we would do the right thing at restore time, instead of reusing the older minTTL which may be too high or inappropriate. This does mean changing updateFromCache to use this new field, instead of the current `.ttl` field.

Reply:
spoked offline about this, will do this in a future PR.